vra/resource_deployment.go

```diff
@@ -370,17 +458,218 @@ func resourceDeploymentUpdate(d *schema.ResourceData, m interface{}) error {
-	return resourceDeploymentRead(d, m)
+	d.SetPartial("name")
+	d.SetPartial("description")
+	log.Printf("Finished updating deployment name and description")
+	return nil
+}
+
+func runDeploymentUpdateAction(d *schema.ResourceData, apiClient *client.MulticloudIaaS, deploymentUuid strfmt.UUID) error {
+	log.Printf("Noticed changes to inputs. Starting to update deployment with inputs")
+	// Get the deployment actions
+	deploymentActions, err := apiClient.DeploymentActions.GetDeploymentActionsUsingGET(deployment_actions.
+		NewGetDeploymentActionsUsingGETParams().WithDepID(deploymentUuid))
+	if err != nil {
+		return err
+	}
+
+	updateAvailable := false
+	actionId := ""
+	for _, action := range deploymentActions.Payload {
+		if strings.Contains(strings.ToLower(action.ID), strings.ToLower("Update")) {
+			// Update action is available on the deployment
+			updateAvailable = true
+			actionId = action.ID
+			break
+		}
+	}
+
+	name := d.Get("name")
+	if !updateAvailable {
+		log.Printf("Update action is not available on the vra_deployment %s, hence new inputs are not applied", name)
+	} else {
+		var inputs = make(map[string]interface{})
+		blueprintId, catalogItemId := "", ""
+		if v, ok := d.GetOk("catalog_item_id"); ok {
+			catalogItemId = v.(string)
+		}
+
+		if v, ok := d.GetOk("blueprint_id"); ok {
+			blueprintId = v.(string)
+		}
+
+		// If catalog_item_id is provided, get the catalog item schema deployment with the catalog item
+		if catalogItemId != "" {
+			catalogItemVersion := ""
+			if v, ok := d.GetOk("catalog_item_version"); ok {
+				catalogItemVersion = v.(string)
+			}
+
+			if v, ok := d.GetOk("inputs"); ok {
+				inputsSchemaMap, err := getCatalogItemSchema(apiClient, catalogItemId, catalogItemVersion)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("Inputs Schema: %v", inputsSchemaMap)
+				inputTypesMap, err := getInputTypesMapFromCatalotItemSchema(inputsSchemaMap)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("InputTypesMap: %v", inputTypesMap)
+				inputs, err = getInputsByType(v.(map[string]interface{}), inputTypesMap)
+				if err != nil {
+					return err
+				}
+			}
+		} else if blueprintId != "" {
+			blueprintVersion := ""
+			if v, ok := d.GetOk("blueprint_version"); ok {
+				blueprintVersion = v.(string)
+			}
+
+			if v, ok := d.GetOk("inputs"); ok {
+				inputsSchemaMap, err := getBlueprintSchema(apiClient, blueprintId, blueprintVersion)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("Inputs Schema: %v", inputsSchemaMap)
+				inputTypesMap, err := getInputTypesMapFromBlueprintInputsSchema(inputsSchemaMap)
+				if err != nil {
+					return err
+				}
+
+				log.Printf("InputTypesMap: %v", inputTypesMap)
+				inputs, err = getInputsByType(v.(map[string]interface{}), inputTypesMap)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		reason := "Updated deployment inputs from vRA provider for Terraform."
```

Comment:
Are we hard coding the reason?

Reply:
Yes, the reason here is different from the deployment description and specific to the action being requested. It is hard-coded so that user may see it in the event history of the deployment instead of no reason.
There already is a `description` in `vra_deployment` resource that captures the deployment description and it gets updated outside of the deployment update action.